File: src/bindings/good_manners.cpp
#pragma once

#include "./good_manners.h"
#include <iostream>

GoodManners::GoodManners()
{
    std::cout << "Hello C++!\n";
}

void GoodManners::please()
{
    std::cout << "Please C++!\n";
}

GoodManners::~GoodManners()
{
    std::cout << "Baye C++!\n";
}

void say_hello()
{
    std::cout << "Hello From C++\n";
}

File: src/bindings/good_manners.nim
{.passC: "-std=c++11 -Isrc/bindings".}

{.compile: "./good_manners.cpp".}

type
    GoodManners* {.importcpp: "GoodManners",header: "./good_manners.h".} = object

proc constructGoodManners*() : GoodManners {.importcpp: "GoodManners()",constructor,header: "./good_manners.h".}
proc please*(this: GoodManners) {.importcpp: "#.please()".}

proc say_hello*() : void {.importcpp: "say_hello()",header: "./good_manners.h".}

